robot_modeling/DQ_SerialManipulator.m

```diff
@@ -158,54 +152,6 @@ function set_effector(obj,effector)
             end
         end
         
-        function dq = dh2dq(obj,theta,i)
-            %   For a given link's DH parameters, calculate the correspondent dual
-            %   quaternion
-            %   Usage: dq = dh2dq(theta,i), where
-            %          theta: joint angle
-            %          i: link number
-            
-            if nargin ~= 3
-                error('Wrong number of arguments. The parameters are theta and the correspondent link')
-            end
-            d = obj.d(i);
-            a = obj.a(i);
-            alpha = obj.alpha(i);
-            
-            if strcmp(obj.convention,'standard')
-                h(1) = cos((theta+obj.theta(i))/2)*cos(alpha/2);
-                h(2) = cos((theta+obj.theta(i))/2)*sin(alpha/2);
-                h(3) = sin((theta+obj.theta(i))/2)*sin(alpha/2);
-                h(4) = sin((theta+obj.theta(i))/2)*cos(alpha/2);
-                d2 = d/2;
-                a2 = a/2;
-                
-                
-                h(5) = -d2*h(4)-a2*h(2);
-                h(6) = -d2*h(3)+a2*h(1);
-                h(7) = d2*h(2)+a2*h(4);
-                h(8) = d2*h(1)-a2*h(3);
-            else
-                h1 = cos((theta+obj.theta(i))/2)*cos(alpha/2);
-                h2 = cos((theta+obj.theta(i))/2)*sin(alpha/2);
-                h3 = sin((theta+obj.theta(i))/2)*sin(alpha/2);
-                h4 = sin((theta+obj.theta(i))/2)*cos(alpha/2);
-                h(1) = h1;
-                h(2) = h2;
-                h(3) = -h3;
-                h(4) = h4;
-                d2 = d/2;
-                a2 = a/2;
-                
-                h(5) = -d2*h4-a2*h2;
-                h(6) = -d2*h3+a2*h1;
-                h(7) = -(d2*h2+a2*h4);
-                h(8) = d2*h1-a2*h3;
-            end
-            
-            dq = DQ(h);
-        end
-        
         function p = get_z(~,h)
```

Comment:
The implementation of this method is specific to the DH convention, although the method signature is general. Therefore, leave it as an abstract method, but move the concrete implementation to the `DQ_SerialManipulatorDH`.

Also, add the following to the (concrete) method implementation:

Eq. (2.34) of Adorno, B. V. (2011). <i>Two-arm Manipulation: From Manipulators to Enhanced Human-Robot Collaboration [Contribution à la manipulation à deux bras : des manipulateurs à la collaboration homme-robot]</i>. https://tel.archives-ouvertes.fr/tel-00641678/

Reply:
- [x] Implemented.